h2o-core/src/main/java/water/rapids/ast/prims/mungers/AstGroup.java

```diff
@@ -753,15 +796,31 @@ public G(int ncols, AGG[] aggs, boolean hasMedian) {
     }
 
     public G fill(int row, Chunk chks[]) {
-      for (int c = 0; c < chks.length; c++) // For all selection cols
-        _gs[c] = chks[c].atd(row); // Load into working array
+      for (int c = 0; c < chks.length; c++) { // For all selection cols
+        Vec vec = chks[c].vec();
+        
+        // Load into working array
+        if (vec.isString())
+          _gsStr[c] = chks[c].stringAt(row);
```

Comment:
it is better to keep everything in BufferedStrings as much as possible (atStr method), this won't duplicate the memory needed to hold the string, BufferedString just references an existing byte array that holds the actual data

Reply:
Makes sense. I will adjust it.

Reply:
I changed it to use `BufferedString` internally.